app/controllers/search_label.js

```diff
@@ -1,96 +1,159 @@
+          }
+        });
+        Ember.run.later(this, this.highlight);
+      } else {
+        types = types.concat(this.get("allPackageTypes").toArray());
+        this.clearHiglight();
+      }
+
+      return types.sortBy("name").uniq();
+    }
+  ),
 
   highlight() {
-    var string = Ember.$.trim(this.get('filter').toLowerCase());
+    var string = Ember.$.trim(this.get("filter").toLowerCase());
     this.clearHiglight();
-    Ember.$(".item_types_result li div").each(function () {
+    Ember.$(".item_types_result li div").each(function() {
       var text = Ember.$(this).text();
-      if((text.toLowerCase()).indexOf(string.toLowerCase()) > -1) {
-        var matchStart = text.toLowerCase().indexOf("" + string.toLowerCase() + "");
+      if (text.toLowerCase().indexOf(string.toLowerCase()) > -1) {
+        var matchStart = text
+          .toLowerCase()
+          .indexOf("" + string.toLowerCase() + "");
         var matchEnd = matchStart + string.length - 1;
         var beforeMatch = text.slice(0, matchStart);
         var matchText = text.slice(matchStart, matchEnd + 1);
         var afterMatch = text.slice(matchEnd + 1);
-        Ember.$(this).html(beforeMatch + "<em>" + matchText + "</em>" + afterMatch);
+        Ember.$(this).html(
+          beforeMatch + "<em>" + matchText + "</em>" + afterMatch
+        );
       }
     });
   },
 
   clearHiglight() {
-    Ember.$("em").replaceWith(function() { return this.innerHTML; });
+    Ember.$("em").replaceWith(function() {
+      return this.innerHTML;
+    });
+  },
+
+  packageParams() {
+    const item = this.get("model");
+    const packageTypeId = item.get("packageType.id");
+    const pkgType = this.get("store").peekRecord("packageType", packageTypeId);
+    return {
+      notes: item.get("packageType.name"),
+      quantity: 1,
+      packageTypeId,
+      packageType: pkgType,
+      offerId: item.get("offer.id"),
+      item: item
+    };
+  },
+
+  addPackage() {
+    let pkgRecord = this.store.createRecord("package", this.packageParams());
+    this.runTask(
+      pkgRecord.save().then(pkg => {
+        this.transitionToRoute("receive_package", pkg.id, {
+          queryParams: { isUnplannedPackage: true }
+        });
+      })
+    );
   },
 
   actions: {
     clearSearch(isCancelled) {
-      this.set('filter', '');
-      this.set('searchText', '');
-      this.set('fetchMoreResult', true);
-      if(!isCancelled) { Ember.$("#searchText").focus(); }
+      this.set("filter", "");
+      this.set("searchText", "");
+      this.set("fetchMoreResult", true);
+      if (!isCancelled) {
+        Ember.$("#searchText").focus();
+      }
     },
 
     cancelSearch() {
       Ember.$("#searchText").blur();
       this.send("clearSearch", true);
       var item = this.get("model");
-      this.transitionToRoute("review_item.accept", item);
+      if (this.get("isUnplannedPackage")) {
+        this.runTask(
+          item.destroyRecord().then(() => {
```

Comment:
Why destroying an item is part of cancelSearch function?
I think different function would better for this, probably `deleteItemAndCancelSearch`.

What do you think?